init commit

Lemonnik opened the pull request.

Lemonnik committed:
init commit

karl-crl commented on DistMult.py:
```diff
@@ -0,0 +1,40 @@
+from torch import nn
+
+
+class DistMult(nn.Module):
```
Worth adding docstring. There are several styles for doc-srtings. IMO, numpy-style is great (https://numpydoc.readthedocs.io/en/latest/format.html)

karl-crl commented on DistMult.py:
```diff
@@ -0,0 +1,40 @@
+
+    def get_embeddings(self, entities: list, embedding_type='entity'):
+        '''
+        entities -- те индексы, которые нас интересуют 
```
Only English in comments)

karl-crl commented on NFM.py:
```diff
@@ -0,0 +1,92 @@
+        # self.pretrain_FM = pretrain_FM
+
+        self.embeddings = nn.Embedding(num_features, num_factors)
+        self.biases = nn.Embedding(num_features, 1)
```
No need to add bias manually, `nn.Linear()` have arg `bias`

karl-crl commented on NFM.py:
```diff
@@ -0,0 +1,92 @@
+        FM = self.prediction(FM)                               # линейный слой предсказаний
+
+        # bias addition
+        feature_bias = self.biases(features)
```
Are biases w_i from article?
<img width="303" alt="Screen Shot 2022-04-28 at 11 31 29 AM" src="https://user-images.githubusercontent.com/17817026/165711573-67d745aa-8d0b-4334-ad61-e2281d42b3cb.png">

karl-crl commented on main.py:
```diff
@@ -0,0 +1,157 @@
+    print("Starting training DistMult...")
+    model.train()
+    davis.return_type = 'ind'
+    loader = DataLoader(davis, batch_size=BATCH_SIZE)
```
You can place dataloader on device, avoiding moving drugs and targets at every batch

Lemonnik committed:
Major fixes

Added argument parsing via Hydra
Added dependencies (poetry file : pyproject.toml)
Added docstrings and comments
Added little wandb logging

Lemonnik committed:
config moved to config/

Lemonnik committed:
Datasets moved to separate file. Drug/target encoding functions are now functions and not class methods

Lemonnik committed:
added new dataset (bindingDB)

Lemonnik committed:
Now drugs and targets are encoded using LabelEncoder from sklearn

karl-crl commented on DistMult.py:
```diff
@@ -0,0 +1,90 @@
+        Parameters
+        ----------
+        n_nodes : int
+            Number of nodes in knoledge graphs. 
```
Spelling error: kno**w**ledge

karl-crl commented on DistMult.py:
```diff
@@ -0,0 +1,90 @@
+        n_relations : int
+            Number of interactions between proteins and drugs.
+        embedding_dim : int
+            Embedding dimension. Hyperparameter.
```
Add type `int`

karl-crl commented on DistMult.py:
```diff
@@ -0,0 +1,90 @@
+        self.n_nodes = n_nodes
+        self.n_relations = n_relations
+        self.embedding_dim = embedding_dim
+        # TODO: add option to work with graphs, where heads and tail are different kind of entities
```
It can differs but all type of nodes precess the same way

karl-crl commented on DistMult.py:
```diff
@@ -0,0 +1,90 @@
+        tail_indices : torch.tensor[batch_size]
+            Tensor containing all tail indicices,
+            e.g. encoded drugs/proteins.
+        relation_indices : torch.tensor[batch_size]
```
Not "exist or not", index should specify relation type. In DTI task we have only one type, on other tasks we can have multiple relationship types (eg, different side effects on poly pharmacy task)

karl-crl commented on NFM.py:
```diff
@@ -0,0 +1,124 @@
+
+    def forward(self, features, feature_values):
+        '''
+        Predicts a relation based on vactor of feature values.
```
Typo: v**e**ctor

karl-crl commented on datasets.py:
```diff
@@ -0,0 +1,174 @@
+        '''
+        # Load dataset
+        if df == 'davis':
+            dataFolder = 'https://raw.githubusercontent.com/kexinhuang12345/MolTrans/master/dataset/DAVIS/'
```
Better to move paths to some separate file

karl-crl commented on main.py:
```diff
@@ -0,0 +1,180 @@
+    # Load dataset
+    davis = LoadDavis()
+
+    wandb.init(project="gpt-3")
```
Change project name

karl-crl commented on main.py:
```diff
@@ -0,0 +1,180 @@
+    # Define SEED and DEVICE
+    torch.manual_seed(cfg.run_args.seed)
+    if cfg.run_args.gpu:
+        DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
```
Add print or log device that is used

karl-crl commented on main.py:
```diff
@@ -0,0 +1,180 @@
+    davis.add_feature('embeddings', embs.tolist())
+
+    # total_features = DISTMULT_DIM*2 (heads + tails) + 2124
+    # (2124 = 1023 drug features in Morgan Fingerprint + 101 protein features)
```
No magic constants!

karl-crl commented on main.py:
```diff
@@ -0,0 +1,180 @@
+        DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+
+    # Initialize KGE model
+    model = DistMult(davis.n_entities, davis.n_relations, cfg.model.kge.embed_dim)
```
In DistMult "number of relations" means "number of different types of relations". So in DTI case it will be one type

karl-crl commented on main.py:
```diff
@@ -0,0 +1,180 @@
+    model_nfm = model_nfm.to(DEVICE)
+    
+    # TRAINING CYCLE
+    print("\nStarting training NFM...")
```
Repeating code

Lemonnik committed:
utils.py file is now correct

Lemonnik committed:
fixes according to code review

Lemonnik committed:
Update Readme

Project description updated

Lemonnik committed:
small changes

small changes

Lemonnik committed:
Update README.md

Lemonnik committed:
Update README.md

Lemonnik committed:
models moved to models/ dir

Lemonnik committed:
final fixes
